### PR TITLE
:scroll: Add editorconfig file. [skip ci]

### PR DESCRIPTION
**Pull request by kaikai1024**
#### Description of the Change

A editorconfig file

#### Applicable Issues

#404

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+charset = utf-8
+indent_style = space
+indent_size = 4
+
+# Matches multiple files with brace expansion notation
+# Uncomment these when changing the default indent config
+# [*.{rs,json,py,md,markdown}]
+# indent_style = space
+# indent_size = 4
+
```
Comment (ouwenkg) [inline .editorconfig]: why uncomment these lines?

Comment (ouwenkg) [inline .editorconfig]: https://github.com/editorconfig/editorconfig/wiki/Projects-Using-EditorConfig

A little confused about your comments, explain more?

Comment (kaikai1024) [reply .editorconfig]: Check this: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties

Why i comment these lines?

check num 13-14:
```
indent_style = space
indent_size = 4
```

we already have the default indent config and it's same as the comment ones.
but if we change the indent config in the future(maybe indent_size = 2), while `{rs,json,py,md,markdown}` these should be stay the old config(indent_size =4).

If you config the indent on then `[]` part, use the config. if not use the default at the header of file.

Comment (ouwenkg) [reply .editorconfig]: Ok，just a template confit usage.

Comment (kaikai1024) [reply .editorconfig]: It's better to uncomment these in case of changing the default config. I will fix it later

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+# Set default charset
+charset = utf-8
+indent_style = space
+indent_size = 4
+
+# Follow https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md
```
Comment (yangby-cryptape) [inline .editorconfig]: We already use CI (rust-fmt) to ensure this.

Comment (kaikai1024) [reply .editorconfig]: yes.

It's better to have this or you need to run the rustFmt

**Pull request merged**